dask-gateway-server/dask_gateway_server/backends/base.py

```diff
@@ -320,6 +346,68 @@ class ClusterConfig(Configurable):
         config=True,
     )
 
+    cluster_max_memory = MemoryLimit(
+        None,
+        help="""
+        The maximum amount of memory (in bytes) available to this cluster.
+        Allows the following suffixes:
+
+        - K -> Kibibytes
+        - M -> Mebibytes
+        - G -> Gibibytes
+        - T -> Tebibytes
+
+        Set to ``None`` for no limit (default).
+        """,
+        min=0,
+        allow_none=True,
+        config=True,
+    )
+
+    cluster_max_cores = Float(
+        None,
+        help="""
+        The maximum number of cores available to this cluster.
+
+        Set to ``None`` for no limit (default).
+        """,
+        min=0.0,
+        allow_none=True,
+        config=True,
+    )
+
+    cluster_max_workers = Integer(
+        help="""
+        The maximum number of workers available to this cluster.
+
+        By default this is computed from ``cluster_max_memory`` and
+        ``cluster_max_cores``.
+
+        Set to ``None`` for no limit (default).
+        """,
+        allow_none=True,
+        min=0,
+        config=True,
+    )
+
+    @default("cluster_max_workers")
+    def _default_cluster_max_workers(self):
```

Comment:
If `cluster_max_workers` is explicitly set, then `cluster_max_cores`/`cluster_max_memory` are ignored, even if those restrictions are stricter. I'm waffling on whether that's the right behavior. Right now I think that's confusing, and we should take the min of the 3.

Reply:
I'm +1 on taking the min of the 3. That seems more intuitive